### DISCUSSION TITLE
Cannot use Crazyradio PA to connect

### DISCUSSION THREAD

**Discussion by Thijxx** [bitcraze/discussions#692] [Q&A] [answered]
When I try to connect to the Crazyflie rev. D using the Crazyradio PA rev. B it hangs on Connecting... and the terminal outputs some interesting things.

- Power-on the Crazyflie
- Plug in the Crazyradio
- Start cfclient
- Scan button in cfclient
- Pick radio 0/80/250K and press Connect

Connecting over Bluetooth and flying the machine works.
I do not know the firmware versions of the dongle or the crazyflie.

Because the USB seems to give an unexpected output I tried to flash the firmware of the Crazyradio first, it responded with `bcdDevice            0.53` after I executed `lsusb -d 1915:7777 -v | grep bcdDevice` so this looks ok.

```
WARNING:cflib.crtp.usbdriver:Exception while scanning for Crazyflie USB: 'NoneType' object has no attribute 'bcdDevice'
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
INFO:cflib.crazyflie:Callback->Connection initialized[radio://0/80/250K]
INFO:cflib.crazyflie:We are connected[radio://0/80/250K], request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
INFO:cflib.crazyflie:Callback->Connected to [radio://0/80/250K]
INFO:cflib.crazyflie.platformservice:_crt_service_callback
ERROR:cflib.crazyflie:Exception while doing callback on port [15]

Traceback (most recent call last):
  File "/home/thijs/.local/lib/python3.10/site-packages/cflib/crazyflie/__init__.py", line 415, in run
    cb.callback(pk)
  File "/home/thijs/.local/lib/python3.10/site-packages/cflib/crazyflie/platformservice.py", line 108, in _crt_service_callback
    if pk.data[:18].decode('utf8') == 'Bitcraze Crazyflie':
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbe in position 8: invalid start byte

```

**Comment by ataffanel** [comment]
What version of the Python, Crazyflie firmware and the client are you using? This crashes when trying to detect the version of the firmware (one radio exchange should return a packet containing `'Bitcraze Crazyflie'`, In that case the radio packet contains a wrong string which leads me to thing that one element is not running the latest version (or a recent enough version at least).

It does seems that the radio is working fine though! :-).

I suggest you update the Crazyflie firmware using the cold-boot method (menu connect/bootloader -> cold boot tab). You need to start the Crazyflie by pressing the button until the blue LED starts blinking to put it in bootloader mode, both blue LED should blink when in bootloader mode.

**Comment by Thijxx** [comment]
I do not know the firmware version of the crazyflie.
What Python version cfclient uses I do not know, I cloned the crazyflie-clients-python and did the pip3 install described in the Installation Docs, a Python3 version I suppose.

I have no cable connected to the Flie, was somewhatv expecting to use a cable to flash firmware.
When I press the button for 3 seconds there is only one LED blinking, I have tried pressing it longer.

When I Scan for the device in the main screen and then go to the Bootloader window the process hangs on this:
![image](https://github.com/bitcraze/discussions/assets/41118940/d8c5a65e-6ba2-42a3-9846-4fb7322c14c8)

I do see some data in the background and on the Flie there is a green LED blinking like crazy (M4) and a red LED slowly blinking on M1. Both other LED's are continues blue.

![image](https://github.com/bitcraze/discussions/assets/41118940/959ef937-ef13-43e5-865c-524057cbc788)

**Reply by tobbeanton** [reply]
Possibly the radio MCU firmware is corrupt. Try to enter bootloader mode from an un-powered state. That is, hold the button and then connect the battery.

**Reply by Thijxx** [reply, marked as answer]
M2 goes from slowly blinking to fast blinking and that's it.

I updated it using iOS.
Now I can connect to it but I don't get telemetry, maybe I need to set things up for that first.
